### DISCUSSION TITLE
NOT LIKE Operator for Filters or something similar

### DISCUSSION THREAD

**Discussion by kemp12322** [meilisearch/product#619] [Feedback & Feature Proposal]
Since we already have a NOT operator, is it possible to add a NOT LIKE operator or something similar?

I have a search functionality that I need to implement.
It has a functionality to get all items that does not contain a dynamic keyword.

Is there a way to implement this in our latest version?

if not maybe we can look at implementing something like this:

```
client.index('movies').search('operator:NOT keyword:Batman')

// OR

client.index('movies').search('', {
  filter: ['NOT LIKE name = 'Batman']
})
```

so basically this gets all the movies that does not contain "Batman".
lets say we have this array
```
[
  {
     name: 'The Dark Knight Rises'
  },
  
  {
     name: 'Batman Begins'
  },

  {
     name: 'The Batman'
  },
]
```

The result would only return:

```
[
  {
     name: 'The Dark Knight Rises'
  }
]
```

What do you guys think?

**Comment by gmourier** [comment]
Hello @kemp12322 👋

We plan to explore a `CONTAINS/LIKE` operator soon. The NOT should be natively supported when we will add it.

Please, follow that [discussion](https://github.com/meilisearch/product/discussions/544) if you want to be updated about it.
